### PR TITLE
JAVA-5415: add artifact tracing on publish

### PR DESCRIPTION
**Pull request by dylrich**
This commit adds papertrail tracing on release to all modules we release. This tracing will run on every snapshot and release.

### CODE REVIEW

File: gradle/publish.gradle [hunk 1]
```diff
@@ -74,6 +74,13 @@ def scalaProjects = publishedProjects.findAll { it.name.contains('scala') }
 def javaProjects = publishedProjects - scalaProjects
 def projectsWithManifest = publishedProjects.findAll {it.name != 'driver-legacy' }
 
+configure(publishedProjects) { project ->
```
[the remark this answers is not in the document]
Comment (dylrich) [reply gradle/publish.gradle:77]: I updated the commit message to include more of this context!

**Comment by dylrich** [comment]
(I don't believe the failures in CI have anything to do with this change)

File: gradle/publish.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (rozza) [reply gradle/publish.gradle:77]: I recommend removing this block and instead adding this to line 57:

```
maven {
    url = "$rootDir/build/repo"
}
```

This change then publishes to two locations: sonatype and the local root `build/repo` directory.  You can then copy that directory for artifact tracing.

**Review by rozza — CHANGES_REQUESTED** [review]

File: gradle/publish.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (dylrich) [reply gradle/publish.gradle]: I have made this change and updated the commit message to reflect this approach instead. This seems much better!

File: .evergreen/.evg.yml [hunk 2]
```diff
@@ -821,6 +821,40 @@ functions:
...
+      params:
+        file: src/trace-expansions.yml
+    - command: papertrail.trace
+      params:
+        key_id: ${papertrail_key_id}
+        secret_key: ${papertrail_secret_key}
+        product: ${product}
+        version: ${release_version}
+        filenames:
+          - "src/build/repo/org/mongodb/*/*/*.jar"
```
Comment (jyemin) [inline .evergreen/.evg.yml:848]: Could all these be replaced with one line, something like:

`"src/build/repo/org/mongodb/*/*/*"`

?

Comment (dylrich) [reply .evergreen/.evg.yml:848]: Unfortunately not -- I discovered that there are multiple files (with the same name) included in these directories that document what we pushed, but are not considered release artifacts. They do not show up on in the remote repos, and are some kind of metadata xml file. I agree it'd be nicer to read if we could, though

**Review by jyemin — APPROVED** [review]
LGTM

**Review by rozza — APPROVED** [review]
LGTM

**Comment by dylrich** [comment]
The CI failures seem unrelated -- is this OK to merge now? I myself don't have permission

**Comment by rozza** [comment]
@dylrich merged 👍

**Pull request merged**